phosphor-auditlog/auditLogMgr.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include "auditLogUtils.hpp"
+
+#include <libaudit.h>
+
+#include <phosphor-logging/lg2.hpp>
+#include <sdbusplus/bus.hpp>
+#include <sdbusplus/server/object.hpp>
+#include <sdeventplus/event.hpp>
+#include <sdeventplus/source/event.hpp>
+#include <xyz/openbmc_project/Common/File/error.hpp>
+#include <xyz/openbmc_project/Logging/AuditLog/server.hpp>
+
+#include <string>
+
+namespace phosphor
+{
+namespace auditlog
+{
+
+using ALIface = sdbusplus::xyz::openbmc_project::Logging::server::AuditLog;
+using ALObject = sdbusplus::server::object_t<ALIface>;
+
+/** @class ALManager
+ *  @brief Configuration for AuditLog server
+ *  @details A concrete implementation of the
+ *  xyz.openbmc_project.Logging.AuditLog API, in order to
+ *  provide audit log support.
+ */
+class ALManager : public ALObject
+{
+  public:
+    ALManager() = delete;
+    ALManager(const ALManager&) = delete;
+    ALManager& operator=(const ALManager&) = delete;
+    ALManager(ALManager&&) = delete;
+    ALManager& operator=(ALManager&&) = delete;
+    virtual ~ALManager() = default;
+
+    /** @brief Constructor to put object onto bus at a dbus path.
+     *  @param[in] bus - Bus to attach to.
+     *  @param[in] path - Path to attach at.
+     */
+    ALManager(sdbusplus::bus_t& bus, const std::string& path) :
+        ALObject(bus, path.c_str()){};
+
+    sdbusplus::message::unix_fd parseAuditLog() override;
+    sdbusplus::message::unix_fd getAuditLog(uint32_t maxCount) override;
+
+  private:
+    int openParseFD(ALParseFile& parsedFile);
```

Comment:
missing doxygen

Reply:
Sorry for missing these. I've gone through and added for all of the added files.

I had some of the doxygen in the cpp files and some in the hpp files. I've moved the ones from the cpp files to the hpp files where definitions exist.